### ISSUE TITLE
jupyter notebooks in codespaces

### ISSUE THREAD

**Issue by RemDelaporteMathurin** [jorgensd/dolfin_pyvista_adapter#1]
Hi!

Thanks for this package!

I can't manage to run [this example](https://github.com/jorgensd/dolfin_pyvista_adapter/blob/main/docs/demo.ipynb) in a GitHub codespace.

I've seen in the pyvista doc that [we may have to use jupyter-server-proxy](https://docs.pyvista.org/version/stable/user-guide/jupyter/trame.html#jupyter-server-proxy)

So the full example i tried was:

```python
import pyvista
pyvista.global_theme.trame.server_proxy_enabled = True

import dolfin
from dolfin_pyvista_adapter import plot
mesh = dolfin.UnitCubeMesh(2, 2, 2)
Vv = dolfin.VectorFunctionSpace(mesh, "Lagrange", 3)
v = dolfin.Function(Vv)
v.interpolate(dolfin.Expression(("x[0]", "x[1]", "pow(x[2], 3)"), degree=1))
plot(v, show_edges=True, glyph_factor=0.1, off_screen=False)
```

```
vscode-remote+codespaces-002bremdelaportemathurin-002dobscure-002ddollop-002dx97wjwwpgrq3p494.vscode-resource.vscode-cdn.net’s server IP address could not be found.
```

<img width="761" alt="image" src="https://github.com/jorgensd/dolfin_pyvista_adapter/assets/40028739/7ae5a4f5-f112-4621-ae00-9872dbe82fb8">

I must admit these IP issues are a bit unclear to me.

Do you have any idea?

Cheers
Rem

**Comment by jorgensd** [comment]
You could try to use the `panel` backend instead (as `trame` is a bit of a moving target atm).
```python
%pip install panel
pyvista.set_jupyter_backend('panel')
```

**Comment by RemDelaporteMathurin** [comment]
Yes that worked!

Just had the following deprecation message though:

```
/tmp/ipykernel_18944/954588667.py:4: PyVistaDeprecationWarning: `panel` backend is deprecated and is planned for future removal.
  pyvista.set_jupyter_backend('panel')
```

Thanks!

**Comment by jorgensd** [comment]
> Yes that worked!
> 
> Just had the following deprecation message though:
> 
> ```
> /tmp/ipykernel_18944/954588667.py:4: PyVistaDeprecationWarning: `panel` backend is deprecated and is planned for future removal.
>   pyvista.set_jupyter_backend('panel')
> ```
> 
> Thanks!

Yep, I know it is going to be deprecated soon, which is a bit of a tricky situation.
I'm working on moving all my work to trame, but getting to know the new backend takes a bit of time.

I'll see if I can get around to test this in codespaces at some point this summer

**Comment by RemDelaporteMathurin** [comment]
I know that pyvista recently added a XDMFreader too if that's of interest!

**Comment by jorgensd** [comment]
For me xdmf support is only an option if it supports the xdmf files written by `write_checkpoint`, as these support other function spaces than  first order Lagrange (standard XDMFFile.write or File.write interpolates into P-1 at time of writing).
